Litecactus docs GitHub edit

litecactus opened the pull request:
I have added a few lines to the docs **Users.md** file to better explain how the Github Auth plugin is setup. I spent a while trying to get it setup and thought the docs could do with some more information which would make it easier for someone else. Very minor change to documentation only.

### Check list

> Check if done, if not relevant leave unchecked.

- [ ] PR reference the relevant issue (e.g. `Fixes #007` or `See xoa-support#42`)
- [ ] if UI changes, a screenshot has been added to the PR
- [x] documentation updated
- `CHANGELOG.unreleased.md`:
  - [x] enhancement/bug fix entry added
  - [ ] list of packages to release updated (`${name} v${new version}`)
- **I have tested added/updated features** (and impacted code)
  - [ ] unit tests (e.g. [`cron/parse.spec.js`](https://github.com/vatesfr/xen-orchestra/blob/b24400b21de1ebafa1099c56bac1de5c988d9202/%40xen-orchestra/cron/src/parse.spec.js))
  - [ ] if `xo-server` API changes, the corresponding test has been added to/updated on [`xo-server-test`](https://github.com/vatesfr/xen-orchestra/tree/master/packages/xo-server-test)
  - [ ] at least manual testing

### Process

1. create a PR as soon as possible
1. mark it as `WiP:` (Work in Progress) if not ready to be merged
1. when you want a review, add a reviewer (and only one)
1. if necessary, update your PR, and re- add a reviewer

From [_the Four Agreements_](https://en.wikipedia.org/wiki/Don_Miguel_Ruiz#The_Four_Agreements):

1. Be impeccable with your word.
1. Don't take anything personally.
1. Don't make assumptions.
1. Always do your best.

litecactus committed:
update Users.md with more Github Auth plugin info

I've expanded the Github Auth plugin setup info as I struggled to set it up myself and actually found very little info online about it.

litecactus committed:
Update CHANGELOG.unreleased.md

julien-f commented:
Thank you for your contribution :slightly_smiling_face:

litecactus commented:
It seems like this setup allows public access (with default no permissions) see my forum post here (https://xcp-ng.org/forum/topic/7140/github-auth-plugin-scope) 

Hopefully someone with more knowledge on the Github Auth plugin will be able to chip in and explain how to setup the OAuth app without any public permissions and then I can update this even further. Even as is, its still better than it was before.

Fohdeesha reviewed (approved):
Looks good! no grammar/spelling changes needed

julien-f committed:
Update CHANGELOG.unreleased.md

julien-f commented on docs/users.md:
```diff
@@ -113,12 +113,18 @@ Save the configuration and then activate the plugin (button on top).
 This plugin allows GitHub users to authenticate to Xen-Orchestra.
 
-The first time a user signs in, XO will create a new XO user with the same identifier, without any permissions.
+The first time a user signs in, XO will create a new XO user with the same identifier (i.e. Github name), with "user" permissions. An existing admin will need to apply the appropriate permissions for your environment.
```
- *GitHub* capitalization
- you can use `*word*` to highlight a word in Markdown

litecactus replied:
I have updated this now

julien-f commented on docs/users.md:
```diff
@@ -113,12 +113,18 @@ Save the configuration and then activate the plugin (button on top).
+
+Name your Github application under "Application Name".
+Enter your Xen Orchestra URL (or IP) under "Homepage URL" 
+Add your "Authorization callback URL" (for example, https://homepageUrl/signin/github/callback)
```
Mardown ignore line breaks, a numbered list would make sense here.

litecactus replied:
Added numbered list

litecactus committed:
Update users.md

updated as per requests

litecactus committed:
Update users.md

re-added GitHub capitals

litecactus commented:
Off the back of my forum post, I think it's worth noting that the GitHub auth plug cannot limit the scope of the users who can login. Adding this to another commit...

litecactus committed:
Update users.md

added a slight amendment to indicate that point that any GitHub user can authenticate into XOA with this plugin

Fohdeesha reviewed (approved):
looks good to me
